Raspi/Test/DataToTempCSV.py

```diff
@@ -0,0 +1,92 @@
+import json
+from datetime import datetime
+import os
+import csv
+
+
+class DataToTempCSV:
+    """ Single use object to parse the MQTT data and convert it to a temporary
+    CSV file stored in the current derectory"""
+
+    def __init__(self, msg, module_start_time, module_id):
+        self.msg = msg                  # Raw MQTT data
+        self.data_dict = {}             # Data to be output to a temp CSV
+        self.module_id = module_id      # Module_id eg. M1, M2 or M3
+        self.module_start_time = module_start_time  # Start time of the module
+
+        # Decode the data as utf-8 and load into python dict
+        self.module_data = self.msg.payload.decode("utf-8")
+        self.module_data = json.loads(self.module_data)
+
+        # Determine which type of data to parse
+        if msg.topic.endswith("data"):
+            self.type = "DATA"
+            self.parse_module_data()
+
+        elif msg.topic.endswith("battery"):
+            self.type = "BATTERY"
+            self.parse_module_battery()
+
+        elif msg.topic == "/v3/wireless-module/battery/low":
+            self.type = "BATTERY_LOW"
+            # Nothing to parse
+
+        # Find the difference in seconds to when the recording was started and
+        # when the data was recieved.
+        self.time_delta = datetime.now() - self.module_start_time
+        self.time_delta = self.time_delta.total_seconds()
+        self.data_dict[self.module_id+"_"+self.type+"_TIME"] = self.time_delta
```

Comment:
I would put this into another variable `dict_key = self.module_id+"_"+self.type+"_TIME"`. Don't be afraid to use another line of code if it means the code is easier to read!

Reply:
This would be a good place to use fstrings https://cito.github.io/blog/f-strings/

Reply:
I didn't realise that python had fstrings. I will definitely use them instead as it's hard to read.